### PR TITLE
[backend] feat: send match details

### PR DESCRIPTION
**Pull request by sltsheryl**
(rmb to update .env matching)
- changed question difficulty to be multiselect 
Screenshot 2023-10-01 at 11 32 25 AM

- from matching form, `user id`, `categories` and `complexities` are retrieved 
- the values are inserted into the matching table (`socket id`, `user id`, `categories`, `complexities`, `status`)
- used the same mongodb cluster we had in question-service, but different collection
Screenshot 2023-10-01 at 11 33 19 AM

- haven't done the matching in the table yet + publish the result to the queue

### CODE REVIEW

File: frontend/src/context/UserContext.tsx
Comment (seelengxd) [inline frontend/src/context/UserContext.tsx:1]: duplicate of `selectUser` selector and `setUser` action in `authSlice.ts` (user object will be maintained twice)

File: frontend/src/pages/routing/AuthenticatedApp.tsx [hunk 1]
```diff
@@ -18,7 +20,7 @@ const AuthenticatedApp: React.FC = () => {
       <Route path="/question/:questionId/edit" element={<ProtectedRoute child={<UpdateQuestion />} />} />
 
       {/* Matching routes */}
-      <Route path="/matching" element={<Matching />} />
+      <Route path="/matching" element={<Matching userId={user != null ? user.id : 0} />} />
```
Comment (seelengxd) [inline frontend/src/pages/routing/AuthenticatedApp.tsx]:
```suggestion
      <Route path="/matching" element={<Matching />} />
```

File: frontend/src/pages/matching/Matching.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ enum MatchingStateEnum {
   MATCHED,
 }
 
-const Matching: React.FC = () => {
+const Matching: React.FC<{ userId: number }> = ({ userId }) => {
```
Comment (seelengxd) [inline frontend/src/pages/matching/Matching.tsx]:
```suggestion
const Matching: React.FC = () => {
  const userId = useAppSelector(selectUser)?.id;
```

(this should give the id, so we won't need useContext or passing props)

**Review by seelengxd — COMMENTED** [review]
nice! we can get the user id in <Matching /> directly 👍

File: frontend/src/context/UserContext.tsx
Comment (sltsheryl) [reply frontend/src/context/UserContext.tsx:1]: ahh i see thanks 🙂

**Review by seelengxd — APPROVED** [review]
lgtm! (usercontext currently unused)

**Pull request merged**